### ISSUE TITLE
Add cross validation options to Stochastic Tools

### ISSUE THREAD

**Issue by crb3874** [idaholab/moose#21399]
## Reason
Cross validation is a useful technique for evaluating performance and improving confidence in surrogate models. 

## Design
The implementation will attempt to be minimally intrusive to the current model training work flow. An optional cross validation routine will be added to the beginning of `SurrogateTrainer::execute()` that will perform a similar model training loop to what is already used, with additional logic for shuffling and splitting rows in the `Sampler` into training and testing subsets. It will require a `SurrogateModel` to be specified in the input file for evaluating model predictions for the hold out set. This will make the capability generic to the trainer/model used, with minor alterations needed in `preTrain()` to reset the trainer to its initial state at the beginning of each loop. All of this will be controllable by adding input parameter options to `SurrogateTrainer` that can be easily added to existing input files.

## Impact
This will add generic cross validation features for `SurrogateModels` in Stochastic Tools.
